src/entities/join/parser.ts

```diff
@@ -0,0 +1,25 @@
+import { BaseJoin } from '.';
+import { WeightedJoin } from './weighted/weightedJoin';
+
+/*********************** Basic Helper to get join class from pool type *************/
+export type JoinConfig = {
+    customPoolJoins: Record<string, BaseJoin>;
+};
+
+export class JoinParser {
+    private readonly poolJoins: Record<string, BaseJoin> = {};
+
+    constructor(config?: JoinConfig) {
+        const { customPoolJoins } = config || {};
+        this.poolJoins = {
+            Weighted: new WeightedJoin(),
+            // custom pool Joins take precedence over base Joins
+            ...customPoolJoins,
+        };
+    }
+
+    public getJoin(poolType: string): BaseJoin {
+        // TODO - Need to parse
```

Comment:
Not sure what this TODO was referencing?

Reply:
This was left from the initial draft and I forgot to remove. Will do it now 👍